### PR TITLE
WT-7663 The local store extension allows only readonly FS operations.

### PR DESCRIPTION
**Pull request by ddanderson**

### CODE REVIEW

File: ext/storage_sources/local_store/local_store.c [hunk 1]
```diff
@@ -1083,16 +1019,21 @@ local_size(WT_FILE_SYSTEM *file_system, WT_SESSION *session, const char *name, w
...
-        if ((ret = local_cache_path(file_system, name, &path)) != 0)
+        /* It's not in the cache, try the bucket directory. */
+        free(path);
+        if ((ret = local_bucket_path(file_system, name, &path)) != 0)
             goto err;
-
-        ret = stat(path, &sb);
-        /* TODO: tiered: if we still get an ENOENT, then we'd need to ping the cloud to get the
-         * size. */
+        ret = stat(name, &sb);
```
Comment (keitharnoldsmith) [inline ext/storage_sources/local_store/local_store.c:1036]: It looks like this code is the same as in `local_exist()`. Does it makes sense to refactor into a separate function?

File: ext/storage_sources/local_store/local_store.c [hunk 2]
```diff
@@ -1249,14 +1190,10 @@ local_file_size(WT_FILE_HANDLE *file_handle, WT_SESSION *session, wt_off_t *size
...
-    WT_FILE_HANDLE *wt_fh;
-
-    local_fh = (LOCAL_FILE_HANDLE *)file_handle;
-    wt_fh = local_fh->fh;
-
-    local_fh->local->fh_ops++;
-    return (wt_fh->fh_sync(wt_fh, session));
+    /* This is a no-op.  We could also disallow it. */
+    (void)file_handle;
+    (void)session;
```
Comment (keitharnoldsmith) [inline ext/storage_sources/local_store/local_store.c:1195]: `WT_UNUSED()`?

File: ext/storage_sources/local_store/local_store.c [hunk 3]
```diff
@@ -1268,13 +1205,14 @@ local_file_write(
   WT_FILE_HANDLE *file_handle, WT_SESSION *session, wt_off_t offset, size_t len, const void *buf)
 {
     LOCAL_FILE_HANDLE *local_fh;
-    WT_FILE_HANDLE *wt_fh;
 
-    local_fh = (LOCAL_FILE_HANDLE *)file_handle;
-    wt_fh = local_fh->fh;
+    (void)offset;
+    (void)len;
+    (void)buf;
```
Comment (keitharnoldsmith) [inline ext/storage_sources/local_store/local_store.c:1211]: `WT_UNUSED()` ?

**Review by keitharnoldsmith — APPROVED** [review]
LGTM.  Some minor questions/notes in the comments.

**Comment by ddanderson** [comment]
evergreen retry

**Comment by ddanderson** [comment]
@keitharnoldsmith, can you please re-review?  I refactored the `local_exist` and `local_size` to use a shared runction.
`WT_UNUSED` requires `wt_internal.h`, which I'm trying to avoid from an extension.

File: ext/storage_sources/local_store/local_store.c [hunk 4]
```diff
@@ -1012,36 +1034,17 @@ local_size(WT_FILE_SYSTEM *file_system, WT_SESSION *session, const char *name, w
...
 
-    ret = stat(path, &sb);
-    if (ret == ENOENT) {
-        /* It's not in the cache, try the bucket directory. */
-        free(path);
-        if ((ret = local_bucket_path(file_system, name, &path)) != 0)
-            goto err;
-        ret = stat(name, &sb);
-    }
     if (ret == 0)
```
Comment (keitharnoldsmith) [inline ext/storage_sources/local_store/local_store.c:1045]: `ret` should always be zero here.

**Review by keitharnoldsmith — APPROVED** [review]
SLGTM (still looks good to me)

One minor comment.

**Pull request merged**